src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs

```diff
@@ -116,7 +116,7 @@ public override void UpdateAdapter()
 
 			var oldItemViewAdapter = ItemsViewAdapter;
 
-			if (oldItemViewAdapter != null)
+			if (oldItemViewAdapter != null && _initialized)
```

Comment:
Initializing the CarouselView, the first property invalidated is the ItemsSource property, which is responsible at the Android platform level for creating the Adapter etc. After that, the ItemTemplate property also updates the Adapter that already exists https://github.com/dotnet/maui/blob/5a95a116fe4596d775214c055e8155b15dac19ca/src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs # L122 and resets the position to 0.

With this change, we only change the Adapter if the PlatformView is initialized.

Reply:
Add an UI Test!

Reply:
Added UI Tests.
<img width="250" alt="Captura de pantalla 2023-07-27 a las 12 20 16" src="https://github.com/dotnet/maui/assets/6755973/69a6b1ca-80ec-402a-a986-4014353a3e62">